report/src/main/webapp/app/components/keyboardShortcuts.directive.js

```diff
@@ -183,14 +189,26 @@ define(['angularAMD', 'userSettingsService'], function (angularAMD) {
               '.url-name:not(.is-hidden)').filter(':first').find(
               testUrlSelector);
           if (_.isEmpty(previousTest)) {
-            $previousElement = suiteContainer.find('.test-name');
-            scrollTo($previousElement);
-            $previousElement.click();
-            currentItem.parents('.url-name').removeClass('is-active');
-            suiteContainer.addClass('is-expanded');
+            if (!_.isEmpty($(suiteContainer).prevAll(
```

Comment:
Maybe it would be worth to put following 20 lines into a separate method? `traverseTreeUp` is a little to long now.